dev/TEAL_opcodes.md

```diff
@@ -75,16 +76,9 @@ The 32 byte Y-component of a public key is the last element on the stack, preced
 - Opcode: 0x06 {uint8 curve index}
 - Stack: ..., A: []byte &rarr; ..., X: []byte, Y: []byte
 - decompress pubkey A into components X, Y
-- **Cost**: 650
+- **Cost**:  Secp256k1=650 Secp256r1=2400
 - Availability: v5
```

Comment:
How is the curve specified? What happens if a ve5/v6 program tries to

Reply:
The curve is given as an immediate argument to `ecdsa_pk_decompress`, so the TEAL reads:
```
ecdsa_pk_decompress Secp256k1
```
or
```
ecdsa_pk_decompress Secp256r1
```
However, the use of such immediates is version-gated, as described here:
```
| Index | Name | In | Notes |
| - | ------ | - | --------- |
| 0 | Secp256k1 |      | secp256k1 curve, used in Bitcoin |
| 1 | Secp256r1 | v7  | secp256r1 curve, NIST standard |
```

Since Secp256r1 is v7, it will neither assemble that TEAL, not execute the corresponding bytecode (it would fail at runtime if you put the bytecode together "by hand" in a v5 program).